server.js

```diff
@@ -1,38 +1,287 @@
+});
+
+// Seed the database
+const Track = mongoose.model('Track', {
+  id: Number,
+  trackName: String,
+  artistName: String,
+  genre: String,
+  year: Number,
+  bpm: Number,
+  energy: Number,
+  danceability: Number,
+  loudness: Number,
+  liveness: Number,
+  valence: Number,
+  length: Number,
+  acousticness: Number,
+  speechiness: Number,
+  popularity: Number,
+});
+
+// Avoid duplicates upon restarting
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Track.deleteMany({});
+
+    topMusicData.forEach((item) => {
+      const newTrack = new Track(item);
+      newTrack.save();
+    });
+  };
+  seedDatabase();
+}
+
+// Get all tracks. Optional queries: Filter by trackName, artistName, genre, year, bpm...
+// Examples: /tracks/?year=2016
+// Examples: /tracks/?bpm=80
+
+/**
+ * @swagger
+ /tracks:
+ *   get:
+ *     summary: List all tracks, optional filters by query parameters
+ *     parameters:
+ *      - name: trackName
+ *        in: query
+ *        required: false
+ *        format: string
+ *      - name: artistName
+ *        in: query
+ *        required: false
+ *        format: string
+ *      - name: genre
+ *        in: query
+ *        required: false
+ *        format: string
+ *      - name: year
+ *        in: query
+ *        required: false
+ *        format: string
+ *     tags: 
+ *      - Tracks
+ *     responses:
+ *       200:
+ *         description: Successful response.
+ */
+app.get('/tracks/', async (req, res) => {
+  let tracks = await Track.find(req.query);
+  if (tracks) {
+    // Get tracks for your party (i.e. danceability based on the number you specify).
+    // Test with different values. For example, choose 70 if you're looking for more energetic dances.
+    // Example: /tracks/?danceability=60
+    if (req.query.danceability) {
+      const songsByDanceability = await Track.find().gt(
+        'danceability',
+        req.query.danceability,
+      );
+      tracks = songsByDanceability;
+    }
+    res.json(tracks);
+  } else {
+    res.status(404).json({ error: 'Not found' });
+  }
+});
+
+// Get track by ID
+
+/**
+ * @swagger
+ * /tracks/id/{id}:
+ *   get:
+ *     summary: Get tracks by id
+ *     parameters:
+ *      - name: id
+ *        in: path
+ *        required: true
+ *        description: Get an id from the "tracks" query above. Click on "Try it out" and then fill in the song id.
+ *        format: integer
+ *     tags:
+ *      - Single result
+ *     responses:
+ *       200:
+ *         description: Successful response.
+ */
+app.get('/tracks/id/:id', async (req, res) => {
+  try {
+    const trackById = await Track.findById(req.params.id);
+    if (trackById) {
+      res.json(trackById);
+    } else {
+      res.status(404).json({ error: 'Track not found' });
+    }
+  } catch (err) {
+    res.status(400).json({ error: 'Invalid Id' });
+  }
+});
+
+// Get all songs by Billie Eilish
+
+/**
+ * @swagger
+ * /tracks/artist/billie_eilish:
```

Comment:
I dont know the solution, but it would have been nice to hide the swagger code somehow. The rest of your code is very neat, but the chunks of swagger code make it hard to get an overview.

Reply:
Good point! Yes, it's quite a lot. I'll look into it for future projects 👀